python/across_api/base/footprint.py

```diff
@@ -0,0 +1,100 @@
+# Copyright © 2023 United States Government as represented by the
+# Administrator of the National Aeronautics and Space Administration.
+# All Rights Reserved.
+
+import numpy as np
+
+
+def ra_dec_to_uvec(ra, dec):
+    phi = np.deg2rad(90 - dec)
+    theta = np.deg2rad(ra)
+    x = np.cos(theta) * np.sin(phi)
+    y = np.sin(theta) * np.sin(phi)
+    z = np.cos(phi)
+    return x, y, z
+
+
+def uvec_to_ra_dec(x, y, z):
+    r = np.sqrt(x**2 + y**2 + z**2)
+    x /= r
+    y /= r
+    z /= r
+    theta = np.arctan2(y, x)
+    phi = np.arccos(z)
+    dec = 90 - np.rad2deg(phi)
+    if theta < 0:
+        ra = 360 + np.rad2deg(theta)
+    else:
+        ra = np.rad2deg(theta)
+    return ra, dec
+
+
+def x_rot(theta_deg):
+    theta = np.deg2rad(theta_deg)
+    return np.array(
+        [
+            [1, 0, 0],
+            [0, np.cos(theta), -np.sin(theta)],
+            [0, np.sin(theta), np.cos(theta)],
+        ]
+    )
+
+
+def y_rot(theta_deg):
+    theta = np.deg2rad(theta_deg)
+    return np.array(
+        [
+            [np.cos(theta), 0, np.sin(theta)],
+            [0, 1, 0],
+            [-np.sin(theta), 0, np.cos(theta)],
+        ]
+    )
+
+
+def z_rot(theta_deg):
+    theta = np.deg2rad(theta_deg)
+    return np.array(
+        [
+            [np.cos(theta), -np.sin(theta), 0],
+            [np.sin(theta), np.cos(theta), 0],
+            [0, 0, 1],
+        ]
+    )
```

Comment:
`astropy` has a handy `rotation_matrix` function that could most of this. 
See details here: https://docs.astropy.org/en/stable/api/astropy.coordinates.CartesianRepresentation.html

Example code:
```
from astropy.coordinates.matrix_utilities import rotation_matrix
rotation = rotation_matrix(30 * u.deg, axis='z')
```

Reply:
I've re-written the entire footprint projection analysis with purely astropy Representations and Transformation methods.